### DISCUSSION TITLE
Failing in reproducing the Transmittance Spectrum of a Waveguide Bend

### DISCUSSION THREAD

**Discussion by sgrigno2** [NanoComp/meep#2288] [Q&A] [answered]
Dear all,

I am not able to reproduce the Transmittance Spectrum of a Waveguide Bend. I simply followed the instruction in the Tutorial basis section but I am getting a completely different output especially for the transmittance. I am using the scheme interface and I have tried two different computers getting exactly the same plot (attached below). Any ideas of what I am doing wrong? 
I used the ctl file contained in the example folder without any modification.

  
![test-flux-WG-meep](https://user-images.githubusercontent.com/95596624/198406742-014d2280-c032-492b-adda-04cda49dd0d0.png)

**Comment by oskooi** [comment, marked as answer]
> I am using the scheme interface

If you built Meep from source, have you tried running `make check` to see whether all the tests pass? Unfortunately, the unit tests are only for the C++ and Python interfaces so even if the test suite passes this does not mean the Scheme interface is working (although I see no reason why the Scheme interface should not work when computing basic things like the flux).

Note that the Scheme interface has long been deprecated and has been replaced by the Python interface since Meep [Meep 1.4](https://github.com/NanoComp/meep/releases/tag/v1.4.0) (January 2018). I highly recommend using the Python interface.

**Reply by sgrigno2** [reply]
Scheme doesn't seem to work properly when using flux. I have switched to the python interface and it works perfectly.

**Comment by sgrigno2** [comment]
Hi,

Thanks for the reply and your suggestions. 

Correct: I have built Meep from the source, and everything works fine after running "make check" (I attached the output below). Moreover, I agree with you. I cannot see any reasons why the Scheme interface should not work when computing basic things. I used Meep heavily in the past and never had any issues. I have also launched other meep examples without the flux option, and everything works perfectly. I'll try again, and I'll keep you posted. If, in the meantime, you have any other suggestions, please let me know.

Here the make check output:
/Applications/Xcode.app/Contents/Developer/usr/bin/make  check-TESTS
PASS: aniso_disp
PASS: bench
PASS: bragg_transmission
PASS: convergence_cyl_waveguide
PASS: cylindrical
PASS: dump_load
PASS: flux
PASS: harmonics
PASS: integrate
PASS: known_results
PASS: near2far
PASS: one_dimensional
PASS: physical
PASS: stress_tensor
PASS: symmetry
PASS: three_d
PASS: two_dimensional
PASS: 2D_convergence
PASS: h5test
PASS: pml
PASS: cyl-ellipsoid-ll
PASS: array-slice-ll
PASS: ring-ll
============================================================================
Testsuite summary for meep 1.24.0
============================================================================
# TOTAL: 23
# PASS:  23
# SKIP:  0
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 0
============================================================================
